vector-im/element-x-android
# [Voice messages] Add voice recording UI

## Type of change


## Content

Add UI for recording voice messages.

Includes some framework for later feature development e.g. presenter, tests, preview state provider.

Note this feature is only partially complete and there is no recording functionality. The feature is behind a feature flag enabled through developer options.

## Motivation and context

- Part of https://github.com/vector-im/element-meta/issues/2084

## Screenshots / GIFs

[Screen_recording_20231011_131408.webm](https://github.com/vector-im/element-x-android/assets/4940864/d7c0588e-adaa-4cf5-a4ce-d6736e75fdc7)


## Tests

- Enable voice messages in developer options
- Go to a room
- Tap and hold record button
- Type text and observe record button disappears

## Tested devices

- OS version(s): Android 13

## Checklist
